src/main/java/Group15/View/MuscleSelectionView.java

```diff
@@ -99,37 +99,86 @@ private static ScrollPane createEquipmentSelectorScrollPane() {
-                for (CheckBox checkbox : equipmentCheckBoxes) {
-                    if (checkbox != allCheckbox) checkbox.setSelected(false);
+    private static void createEquipmentCheckboxFunctionality(Button allButton, Button noneButton, List<CheckBox> equipmentCheckBoxes) {
+        allButton.setOnAction(_ -> {
+            for (Equipment equipment : Equipment.values()) {
+                if(!selectedEquipment.contains(equipment)) {
+                    MuscleSelectionView.selectedEquipment.add(equipment);
+
+                } else {
+                    MuscleSelectionView.selectedEquipment.remove(equipment);
+
+                }
+            }
+            for (CheckBox checkBox : equipmentCheckBoxes) {
+                checkBox.setSelected(true);
+            }
+        });
+
+        noneButton.setOnAction(_ -> {
+            for (Equipment equipment : Equipment.values()) {
+                if(selectedEquipment.contains(equipment)) {
+                    MuscleSelectionView.selectedEquipment.remove(equipment);
+
+                } else {
+                    MuscleSelectionView.selectedEquipment.add(equipment);
+
                 }
             }
+            for (CheckBox checkBox : equipmentCheckBoxes) {
+                checkBox.setSelected(false);
+            }
         });
-        for (CheckBox checkbox : equipmentCheckBoxes) {
-            if (checkbox != allCheckbox) {
-                checkbox.setOnAction(_ -> {
-                    if (checkbox.isSelected()) {
-                        allCheckbox.setSelected(false);
-                        MuscleSelectionView.selectedEquipment.remove(Equipment.valueOf(checkbox.getText()));
-                        if (!MuscleSelectionView.selectedEquipment.contains(Equipment.valueOf(checkbox.getText()))) MuscleSelectionView.selectedEquipment.add(Equipment.valueOf(checkbox.getText()));
-                    } else {
-                        MuscleSelectionView.selectedEquipment.remove(Equipment.valueOf(checkbox.getText()));
-                    }
-                });
+
+        for (CheckBox checkBox : equipmentCheckBoxes) {
+            checkBox.setOnAction(_ -> {
+                if(checkBox.isSelected()) {
+                    MuscleSelectionView.selectedEquipment.add(fromTextToEquipment(checkBox.getText()));
+
+                } else {
+                    MuscleSelectionView.selectedEquipment.remove(fromTextToEquipment(checkBox.getText()));
+
+                }
+            });
+        }
+    }
+
+    private static Equipment fromTextToEquipment (String checkBoxText) {
+        if (checkBoxText == null || checkBoxText.isEmpty()) return null;
+
+        if("EZBar".equalsIgnoreCase(checkBoxText)) {
+            return Equipment.EZBar;
+        }
+        if ("SmithMachine".equalsIgnoreCase(checkBoxText)) {
+            return Equipment.SmithMachine;
+        }
+
+        String caseMatchedText = caseMatchText(checkBoxText);
+
+        for (Equipment equipment : Equipment.values()) {
```

Comment:
Just use the valueOf in this, it is much easier to read, than reading your custom caseMatching